tests/test_smoothie_com.py

```diff
@@ -0,0 +1,258 @@
+import asyncio
+import unittest
+from apollo import utils
+
+from apollo.smoothie import SmoothieCom
+
+
+class SmoothieComTest(unittest.TestCase):
+    def setUp(self):
+        self.loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(self.loop)
+
+        self.address = utils.get_free_os_address()
+        self.smc = SmoothieCom(
+            self.address[0],
+            self.address[1],
+        )
+    def tearDown(self):
+        self.loop.stop()
+        self.loop.close()
+
+    def test_send_M114(self):
+        """ TEST SENDING M114 """
+        print('\n\n','='*20,'TESTING M114','-'*20,'\n')
+        smoothie_read_sequence = [
+            'feedback engaged',
+            'ok',
+            'ok {"foo":123}',
+            'feedback disengaged',
+            'ok',
+        ]
+
+        smoothie_write_sequence = [b'M62\r\n', b'M114\r\n', b'M63\r\n']
+
+        self.smc.writer = utils.Writer()
+        self.smc._read = utils.Reader(smoothie_read_sequence)._read
+
+        self.loop.run_until_complete(self.smc.send('M114', None))
+
+        # THIS TESTS DATA SENT MATCHES WHAT IT IS SUPPOSED TO
+        self.assertEqual(self.smc.writer.drain_buffer, smoothie_write_sequence)
+        
+        # TEST USING READ SEQUENCE? ... smoothie_read_sequence just keeps program running?
+
+    def test_send_G0(self):
+        """ TEST SENDING G0 """
+        print('\n\n','='*20,'TESTING G0','-'*20,'\n')
```

Comment:
I'm a little confused as to when we're pushing code for testing and when PR should be submitted. I have some code to test, not production ready, but seems like I have to do a PR for it to get tested. Should I just go ahead with testing and then do PR?